pkg/asset/machines/openstack/machines.go

```diff
@@ -108,9 +118,15 @@ func Machines(clusterID string, config *types.InstallConfig, pool *types.Machine
 	return machines, nil
 }
 
-func generateProvider(clusterID string, platform *openstack.Platform, mpool *openstack.MachinePool, osImage string, az string, role, userDataSecret string, trunkSupport bool, rootVolumeAZ string) (*openstackprovider.OpenstackProviderSpec, error) {
+func generateProvider(clusterID string, platform *openstack.Platform, mpool *openstack.MachinePool, osImage string, az string, role, userDataSecret string, trunkSupport bool, rootVolumeAZ string, subnet string) (*openstackprovider.OpenstackProviderSpec, error) {
```

Comment:
here I wonder if passing subnet is ok or if we want to pass the full FailureDomain.